app/controllers/nonsipp/sharesdisposal/ReportedSharesDisposalListController.scala

```diff
@@ -219,58 +229,64 @@ object ReportedSharesDisposalListController {
     mode: Mode,
     disposals: Map[Max5000, List[Max50]],
     userAnswers: UserAnswers,
-    optYear: Option[String] = None,
-    optCurrentVersion: Option[Int] = None,
-    optPreviousVersion: Option[Int] = None
+    viewOnlyViewModel: Option[ViewOnlyViewModel],
+    schemeName: String
   ): List[ListRow] =
-    disposals
-      .flatMap {
-        case (shareIndex, disposalIndexes) =>
-          disposalIndexes.map { disposalIndex =>
-            val sharesDisposalData = SharesDisposalData(
-              shareIndex,
-              disposalIndex,
-              userAnswers.get(TypeOfSharesHeldPage(srn, shareIndex)).get,
-              userAnswers.get(CompanyNameRelatedSharesPage(srn, shareIndex)).get,
-              userAnswers.get(HowWereSharesDisposedPage(srn, shareIndex, disposalIndex)).get
-            )
+    if (disposals.isEmpty) {
```

Comment:
Do we need to display this block only if it is view only mode. I.e    if (disposals.isEmpty && mode.isViewOnlyMode) {